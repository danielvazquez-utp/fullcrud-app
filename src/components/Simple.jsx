import PropTypes from 'prop-types';

const Simple = ({titulo, subtitulo, nombre}) => {
  return (
    <>
        <h1>{ titulo }</h1>
        <h4>{ subtitulo }</h4>
        <p>{ nombre }</p>
    </>
  )
}

export default Simple;

Simple.propTypes = {
    titulo: PropTypes.string.isRequired,
    subtitulo: PropTypes.string
}