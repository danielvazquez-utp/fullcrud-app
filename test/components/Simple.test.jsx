import { render } from "@testing-library/react";
import Simple from "../../src/components/Simple";

describe('Pruebas generales del componente Simple', () => {

    test('Se realiza comparación con snapshot', () => {

        const { container } = render( <Simple titulo="Hello world" subtitulo="Final de cuatrimestre" /> );
        //console.log(container);
        expect( container ).toMatchSnapshot();
    });

    
    test('Se debe mostrar el título dentro de un <h1>', () => {

        const titulo = "Hola mundo";
        const { container, getByText } = render( <Simple titulo={ titulo } /> );

        expect( getByText(titulo) ).toBeTruthy();

        const h1 = container.querySelector("h1");
        expect( h1.innerHTML ).toContain( titulo );

    });
    

});