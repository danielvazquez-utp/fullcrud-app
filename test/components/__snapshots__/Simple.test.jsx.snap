// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas generales del componente Simple Se realiza comparaci√≥n con snapshot 1`] = `
<div>
  <h1>
    Hello world
  </h1>
  <h4>
    Final de cuatrimestre
  </h4>
  <p />
</div>
`;
